core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -692,19 +693,45 @@ class TimeSeriesShard(val ref: DatasetRef,
   /**
    * This method is to apply column filters and fetch matching time series partitions.
    *
-   * @param filter column filter
+   * @param filters column filter
    * @param labelNames labels to return in the response
    * @param endTime end time
    * @param startTime start time
    * @param limit series limit
    * @return returns an iterator of map of label key value pairs of each matching time series
    */
-  def labelValuesWithFilters(filter: Seq[ColumnFilter],
+  def labelValuesWithFilters(filters: Seq[ColumnFilter],
                              labelNames: Seq[String],
                              endTime: Long,
                              startTime: Long,
+                             querySession: QuerySession,
                              limit: Int): Iterator[Map[ZeroCopyUTF8String, ZeroCopyUTF8String]] = {
-    LabelValueResultIterator(partKeyIndex.partIdsFromFilters(filter, startTime, endTime), labelNames, limit)
+    val metricShardKeys = schemas.part.options.shardKeyColumns
+    val metricGroupBy = deploymentPartitionName +: clusterType +: shardKeyValuesFromFilter(metricShardKeys, filters)
```

Comment:
Lets discuss this in person in more detail.
I have questions around when the namespace for accounting will be identified correctly, and when it will not

Reply:
made `multiple` as default for missed shardKeyValue